### PR TITLE
Deprecate positional request constructor API

### PR DESCRIPTION
**Pull request by wltsmrz**
* abbreviated arguments (src_  / dst_) to requestPathFindCreate() etc. dropped
* requestTxHistory() dropped, requestTransactionHistory() lives on

### CODE REVIEW

File: src/core/orderbook.js [hunk 1]
```diff
@@ -1142,20 +1142,20 @@ OrderBook.prototype.getOffersSync = function() {
 OrderBook.prototype.toJSON =
 OrderBook.prototype.to_json = function() {
   const json = {
-    taker_gets: {
+    gets: {
```
Comment (clark800) [inline src/core/orderbook.js]: Isn't it ambiguous if you don't specify that it is the "taker" as opposed to the "maker"?

File: src/core/remote.js [hunk 2]
```diff
@@ -1121,6 +1121,7 @@ Remote.prototype.requestTransactionEntry = function(options, callback) {
  * @return {Request} request
  */
 
+Remote.prototype.requestTx =
```
Comment (sublimator) [inline src/core/remote.js]: Should update the jsdoc comments too, so they don't show positional arguments. I forgot to update them last time, and opened a subsequent WIP p.r. fixing them up.

File: src/core/orderbook.js (hunk 1 above)
Comment (sublimator) [inline src/core/orderbook.js]: While the ambiguity is unlikely to be a problem the potential is there, and it's also true the RPC arguments as recognized by rippled follow a taker_* pattern, so when you add it all up it seems like we should go with taker_* ?

**Comment by sublimator** [comment]
LGTM

**Pull request merged**